### PR TITLE
[cloud-provider-azure] Add cloud data discoverer

### PR DESCRIPTION
**Pull request by name212**
## Description
Add cloud data discoverer service

## Why do we need it, and what problem does it solve?
Discovery instance types for cluster autoscaler.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-azure
type: feature
summary: Add cloud data discoverer service which get information about available instance types for node groups.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/azure'...

File: modules/030-cloud-provider-azure/images/cloud-data-discoverer/discoverer.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			}
+
+			res = append(res, v1alpha1.InstanceType{
+				Name:     *r.Name,
+				CPU:      resource.MustParse(cpu),
+				Memory:   resource.MustParse(memory + "Gi"),
+				RootDisk: resource.MustParse("0"),
+			})
+
+			res[0].RootDisk.IsZero()
```
Comment (yalosev) [inline modules/030-cloud-provider-azure/images/cloud-data-discoverer/discoverer.go:126]: Seems like you lost something here :) This line of code or some condition

Comment (name212) [reply modules/030-cloud-provider-azure/images/cloud-data-discoverer/discoverer.go]: removed

**Review by yalosev — APPROVED** [review]

**Pull request merged**